benchmarks/units.py

```diff
@@ -86,3 +104,64 @@ def time_quantity_times_quantity():
 
 def time_quantity_ufunc_sin():
     np.sin(q2)
+
+
+class TimeQuantityOpSmallArray:
+    """
+    Operator benchmarks from https://github.com/astropy/astropy/issues/7546
+    for a small Numpy array.
+    """
+    def setup(self):
+        data = np.array([1., 2., 3.])
+        self.data = data * u.g
+        self.out = data.copy()
+
+        # A different but dimensionally compatible unit
+        self.data2 = 0.001 * data * u.kg
+
+    def time_quantity_square(self):
+        self.data ** 2
+
+    def time_quantity_np_square(self):
+        np.power(self.data, 2)
+
+    def time_quantity_np_square_out(self):
+        np.power(self.data, 2, out=self.out)
+
+    def time_quantity_sqrt(self):
+        self.data ** 0.5
+
+    def time_quantity_np_sqrt(self):
+        np.sqrt(self.data)
+
+    def time_quantity_np_sqrt_out(self):
+        np.sqrt(self.data, out=self.out)
+
+    def time_quantity_equal(self):
+        self.data == self.data2
```

Comment:
Note: In the paper, `np.equal` and `operator.eq` are compared separately. I did not do that here. Is that necessary here or is `==` alone sufficient? Same note for all the following operator benchmarks. Please let me know.

Reply:
operator.eq and `==` are equivalent, so there's no reason to test both of those. Going through `np.equal` with short-circuit `__eq__` if that's defined, so that might be different.